### PR TITLE
1928 stateful aws service

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Create a stateful AWSService. Compose and register it in the composition root.

Part of issue #1928

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~ (Vulture fails. Future PRs will resolve this.)
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
    > Tested by running the Island and ensuring there were no exceptions when it starts.
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by shreyamalviya — APPROVED** [review]

File: monkey/monkey_island/cc/services/aws_service.py [hunk 1]
```diff
@@ -15,59 +15,83 @@
...
-def filter_instance_data_from_aws_response(response):
+class AWSService:
+    def __init__(self, aws_instance: AWSInstance):
+        """
+        :param aws_instance: An AWSInstance object representing the AWS instance that the Island is
+                             running on
+        """
+        self._aws_instance = aws_instance
+
+    def island_is_running_on_aws(self) -> bool:
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/aws_service.py:26]: `is_island_running_on_aws`

File: monkey/monkey_island/cc/services/aws_service.py [hunk 2]
```diff
@@ -15,59 +15,83 @@
...
+    """
+    Consume raw instance data from the AWS API and return only those fields that are relevant for
+    Infection Monkey.
+
+    :param raw_managed_instances_info: The output of
+                                       DescribeInstanceInformation["InstanceInformation"] from the
+                                       AWS API
+    :return: A sequence of mappings, where each Mapping represents a managed AWS instance that
+             is accessible from the Island.
+    :rtype: Sequence[Mapping[str, str]]
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/aws_service.py:87]: This comment line is irrelevant, it's better to add it to the actual function definition

File: monkey/monkey_island/cc/services/initialize.py [hunk 3]
```diff
@@ -11,12 +11,12 @@
 
 def initialize_services(data_dir: Path) -> DIContainer:
     container = DIContainer()
+    container.register_instance(AWSInstance, AWSInstance())
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/initialize.py:14]: This doesn't cause a 2 second delay?

File: monkey/monkey_island/cc/services/initialize.py [hunk 4]
```diff
@@ -11,12 +11,12 @@
...
+
     container.register_instance(
         IFileStorageService, DirectoryFileStorageService(data_dir / "custom_pbas")
     )
-
-    # Takes a while so it's best to start it in the background
-    Thread(target=aws_service.initialize, name="AwsService initialization", daemon=True).start()
+    container.register_instance(AWSService, container.resolve(AWSService))
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/initialize.py:19]: Hmm, so this is linked to line 14, right? You can't swap these lines

**Review by VakarisZ — APPROVED** [review]

File: monkey/monkey_island/cc/services/initialize.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/initialize.py:14]: No, the AWSInstance constructor starts the requests on a thread. If you attempt to access its members before they're ready, those calls will block.

File: monkey/monkey_island/cc/services/initialize.py (hunk 4 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/initialize.py:19]: Right

File: monkey/monkey_island/cc/services/aws_service.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/aws_service.py:26]: We talked about not using hungarian notation for functions that return boolean. See https://guardicore.atlassian.net/wiki/spaces/RES/pages/2800353435/Infection+Monkey+Development+Guidelines+DRAFT#Functions%2FMethods-returning-booleans

**Comment by mssalvatore** [comment]
This PR created in error

**Pull request merged**